Learning/Session 2 – Project setup and login test skeleton

MikitaZhyhadla opened the pull request:
- Configured repository rules (branch protection, squash merge, etc.)
- Integrated Prettier and added format script
- Set up Cypress with base URL
- Created initial project structure and test documentation
- Added global test resources: selectors, URLs, requirements, localization, users
- Created login-page UI test skeleton and linked issue for known bug

MikitaZhyhadla committed:
chore: mark learning session 1

MikitaZhyhadla committed:
docs: remove reviewer example from pull request section

MikitaZhyhadla committed:
docs: remove example from pull request section

MikitaZhyhadla committed:
docs: remove example from pull request section

MikitaZhyhadla committed:
docs: update branch rules to reflect repo settings

MikitaZhyhadla committed:
chore: add Prettier config and ignore file

MikitaZhyhadla committed:
chore: add format script to package.json

MikitaZhyhadla committed:
chore: add Cypress folder structure with e2e.js

MikitaZhyhadla committed:
docs: add test writing guideline

MikitaZhyhadla committed:
docs: add naming conventions

MikitaZhyhadla committed:
chore: update strategy, config, and review fixes

MikitaZhyhadla committed:
Resolve merge conflicts using main version

MikitaZhyhadla committed:
docs: update git strategy documentation

MikitaZhyhadla committed:
Configure Prettier: keep .prettierrc.json, remove duplicate .prettierrc, add format script to package.json, apply formatting to project files

MikitaZhyhadla committed:
chore: configure Cypress with baseUrl and support file

MikitaZhyhadla committed:
docs: add test writing guideline

MikitaZhyhadla committed:
docs: add naming conventions guideline

MikitaZhyhadla committed:
Add shared resource files: selectors, qa-users, urls, l10n, requirements

MikitaZhyhadla committed:
Add login page test scaffold with skipped use cases

MikitaZhyhadla committed:
Add TODO with issue link for incorrect password bug

MikitaZhyhadla committed:
feat: define global variables in e2e.js

MikitaZhyhadla committed:
Implement valid login test case with bug reference

MikitaZhyhadla committed:
Add test credentials to qa-users.json and ignore sensitive data

MikitaZhyhadla committed:
Fix imports and register global variables in e2e.js

ivan-zdanovich-citation commented on cypress/integration/ui/login-page.ui.spec.js:
```diff
@@ -0,0 +1,28 @@
+      cy.get(selectors.login.usernameInput).type(envUsers.standard.username);
+      cy.get(selectors.login.passwordInput).type(envUsers.standard.password);
+      cy.get(selectors.login.loginButton).click();
+      cy.url().should('include', '/inventory.html');
```
Would be nice to place actions/interactions within the before block of context block.
Also would be nice to create a simple command for passing credentials into fields that will consume user data.

MikitaZhyhadla replied:
Moved login actions to beforeEach() inside Valid Login context.
Also added custom command cy.loginAs(user) in cypress/support/commands.js.

ivan-zdanovich-citation commented on cypress/integration/ui/login-page.ui.spec.js:
```diff
@@ -0,0 +1,28 @@
+  });
+
+  context.skip('Invalid credentials', () => {
+    // Not implemented yet
```
please provide particular assertions in it blocks and mark them as skipped as well

MikitaZhyhadla replied:
Added it.skip() blocks with specific assertions for invalid credentials cases under context('Invalid credentials').

ivan-zdanovich-citation commented on cypress/integration/ui/login-page.ui.spec.js:
```diff
@@ -0,0 +1,28 @@
+  });
+
+  context('Empty credentials', () => {
+    it.skip('Should show correct error when both fields are empty', () => {
```
Probably usage of gherkin keywords (Geven, When, Then) as well as prefix that point to particular part of UI could help to improve your tests.

MikitaZhyhadla replied:
Updated test description to use gherkin keywords (Given, When, Then) and added UI prefix to clarify the tested area.

IvanZdanovich reviewed (changes requested):
please resolve comments

MikitaZhyhadla commented:
> please resolve comments



> resolve

I will start resolving comments today or in the next few days.

MikitaZhyhadla commented:
> please resolve comments



> resolve

I will start resolving comments today or in the next few days.

MikitaZhyhadla committed:
refactor: update login-page test skeleton with gherkin and beforeEach

MikitaZhyhadla committed:
feat: add custom command cy.loginAs(user)

MikitaZhyhadla committed:
test: split Gherkin steps into separate lines in login test

MikitaZhyhadla commented:
@IvanZdanovich all suggestions implemented. Please review again when convenient

MikitaZhyhadla committed:
feat: add users from qa-users.json to Cypress.env in e2e.js

MikitaZhyhadla committed:
fix: correct user reference to standardUser from qa-users.json

MikitaZhyhadla committed:
fix: update specPattern in Cypress config

MikitaZhyhadla committed:
fix: update selectors and login command to match new structure

MikitaZhyhadla committed:
fix: update selectors structure with loginPage inside selectors object

MikitaZhyhadla commented:
Minor updates after requesting review:

– Added custom command `cy.loginAs()`  
– Updated `e2e.js` to load users via `Cypress.env('users')`  
– Fixed `selectors.js` structure for `loginPage`  
– Updated `login-page.ui.spec.js` to use correct selectors and `standardUser` credentials

Tests are now passing 
Ready for re-review.

IvanZdanovich commented on cypress/support/requirements.js:
```diff
@@ -0,0 +1,5 @@
+export const reqs = {
+  minPasswordLength: 6,
+  maxUsernameLength: 20
+};
```
Verify if this is a genuine business requirement.
If confirmed, ensure appropriate tests are developed and any issues are documented in reports.

MikitaZhyhadla replied:
As per the Learning Session 2, we were required to save constant values in cypress/support/requirements.js.
These values are currently illustrative and not based on a confirmed business requirement.
A note was added to the file. Once the requirements are clarified, I will update or remove them accordingly.

IvanZdanovich commented on .gitignore:
```diff
@@ -16,3 +16,6 @@ yarn-error.log*
 Thumbs.db
+
+# Sensitive test data
+cypress/sensitive-data/qa-users.json
```
Exclude the entire folder containing sensitive data.
Provide example files that reflect the structure of the sensitive data files.
Add an example file to the repository and mark it with an exclamation mark in the .gitignore file.
Remove the file with sensitive information from your PR and the repository.

MikitaZhyhadla replied:
Replaced the real user data file with `qa-users.example.json`.
Excluded the `sensitive-data/` folder in `.gitignore`, keeping only the example for structure reference.

IvanZdanovich commented on cypress/integration/ui/login-page.ui.spec.js:
```diff
@@ -0,0 +1,45 @@
+    });
+
+    it('Then user should be redirected to inventory page', () => {
+      // Given login page
```
1. please make context block title more descriptive, this block should reflect main conditions to check state.
2. we are trying to wright dependant tests within one test file to improve performance of tests, therefor please use before block with testIsolation: false flag.

MikitaZhyhadla replied:
Updated context block title to better describe the login scenario.
testIsolation: false is already used in describe to support dependent test execution.
Let me know if anything else needs to be adjusted.

IvanZdanovich commented on cypress/integration/ui/login-page.ui.spec.js:
```diff
@@ -0,0 +1,45 @@
+    it('Then user should be redirected to inventory page', () => {
+      // Given login page
+      // When user logs in with standard credentials
+      // Then user should be redirected to inventory page
```
Verify that a simple element like title is rendered correctly.

MikitaZhyhadla replied:
Added title visibility check and updated context blocks for clarity. Verified the title rendering as requested.

MikitaZhyhadla committed:
add illustrative constants with note to requirements.js

MikitaZhyhadla committed:
replace sensitive user data with example file and update gitignore

MikitaZhyhadla committed:
restructure login spec for clearer context and test flow

MikitaZhyhadla committed:
add title check and improve login flow structure

MikitaZhyhadla committed:
add title visibility check before login

MikitaZhyhadla commented:
@IvanZdanovich All requested changes have been implemented. Please review again when convenient.

Commit:
Improvement

Commit:
Improvement

Commit:
Improvement

MikitaZhyhadla committed:
update login test structure based on review

MikitaZhyhadla committed:
add env-users.example for test user data

MikitaZhyhadla committed:
update login test to use Cypress.env users object

MikitaZhyhadla committed:
Refactor login test and config: use env-users.json for credentials

MikitaZhyhadla committed:
update git strategy clarify approval of latest commit before merging

MikitaZhyhadla committed:
update README to include Cypress watch mode info

MikitaZhyhadla commented:
The following changes were made after our joint review — @IvanZdanovich please take another look at my pull request Learning session 2 when convenient:
1. Replaced qa-users.json with env-users.json for sensitive test data (local only)
3. Added env-users.example.json to Git for team visibility
5. Refactored login tests to use Cypress.env('users') for credentials
7. Updated .gitignore to properly exclude real user data
9. Clarified approval requirements in git-strategy.md
11. Updated README.md to include Cypress auto-reload (watch mode) info
